libraries/core_libs/consensus/src/transaction_manager/transaction_manager.cpp

```diff
@@ -133,34 +133,62 @@ uint32_t TransactionManager::insertBroadcastedTransactions(const SharedTransacti
 unsigned long TransactionManager::getTransactionCount() const { return trx_count_; }
 
 std::shared_ptr<Transaction> TransactionManager::getTransaction(trx_hash_t const &hash) const {
+  std::shared_lock transactions_lock(transactions_mutex_);
   auto trx_it = transactions_pool_.get(hash);
   if (trx_it.second) {
     return trx_it.first;
   }
   return db_->getTransaction(hash);
 }
 
-void TransactionManager::removeTransactionsFromPool(SharedTransactions const &trxs) {
+void TransactionManager::saveTransactionsFromDagBlock(SharedTransactions const &trxs) {
   // This lock synchronizes inserting and removing transactions from transactions memory pool with database insertion.
   // Unique lock here makes sure that transactions we are removing are not reinserted in transactions_pool_
   std::unique_lock transactions_lock(transactions_mutex_);
-  for (auto const &trx : trxs) {
-    nonfinalized_transactions_in_dag_.emplace(trx->getHash());
-    if (transactions_pool_.erase(trx->getHash())) {
-      LOG(log_dg_) << "Transaction " << trx->getHash() << " removed from trx pool";
+  auto write_batch = db_->createWriteBatch();
+  vec_trx_t trx_hashes;
+  std::transform(trxs.begin(), trxs.end(), std::back_inserter(trx_hashes),
+                 [](std::shared_ptr<Transaction> const &t) { return t->getHash(); });
```

Comment:
1 last minor thing: we agreed to use `const Type&` style but you use everywhere the opposite Type const& style

Reply:
Do you think I should change it everywhere in the transaction_manager.cpp or just in the places I touched. We do have a mixture in the code now. Maybe we should just change this everywhere in the code at once?

Reply:
I think anything you touch is ok, we can change it everywhere in separate PR that will do it throughout the codebase....